core/src/main/java/org/apache/brooklyn/util/core/logbook/opensearch/OpenSearchLogStore.java

```diff
@@ -0,0 +1,227 @@
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.ssl.SSLContextBuilder;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.apache.brooklyn.util.core.logbook.LogbookConfig.BASE_NAME_LOGBOOK;
+
+/**
+ * Implementation for expose log from ElasticSearch to the logbook API.
+ */
+public class OpenSearchLogStore implements LogStore {
+
+    /*
+     # example config for local default implementation
+     brooklyn.logbook.logStore = org.apache.brooklyn.util.core.logbook.opensearch.OpenSearchLogStore
+     brooklyn.logbook.openSearchLogStore.host = https://localhost:9200
+     brooklyn.logbook.openSearchLogStore.index = brooklyn8
+     brooklyn.logbook.openSearchLogStore.user = admin
+     brooklyn.logbook.openSearchLogStore.password = admin
+     brooklyn.logbook.openSearchLogStore.verifySSl = false
+     */
+    public final static String BASE_NAME_OPEN_SEARCH_LOG_STORE = BASE_NAME_LOGBOOK + ".openSearchLogStore";
+
+    public final static ConfigKey<String> LOGBOOK_LOG_STORE_HOST = ConfigKeys.newStringConfigKey(
+            BASE_NAME_OPEN_SEARCH_LOG_STORE + ".host", "Log store host");
+
+    public final static ConfigKey<String> LOGBOOK_LOG_STORE_INDEX = ConfigKeys.newStringConfigKey(
+            BASE_NAME_OPEN_SEARCH_LOG_STORE + ".index", "Log store index");
+
+    public final static ConfigKey<String> LOGBOOK_LOG_STORE_USER = ConfigKeys.newStringConfigKey(
+            BASE_NAME_OPEN_SEARCH_LOG_STORE + ".user", "User name");
+
+    public final static ConfigKey<String> LOGBOOK_LOG_STORE_PASS = ConfigKeys.newStringConfigKey(
+            BASE_NAME_OPEN_SEARCH_LOG_STORE + ".password", "User password");
+
+    public final static ConfigKey<String> LOGBOOK_LOG_STORE_APIKEY = ConfigKeys.newStringConfigKey(
+            BASE_NAME_OPEN_SEARCH_LOG_STORE + ".apikey", "API key");
+
+    public final static ConfigKey<Boolean> LOGBOOK_LOG_STORE_VERIFY_SSL = ConfigKeys.newBooleanConfigKey(
+            BASE_NAME_OPEN_SEARCH_LOG_STORE + ".verifySSl", "Verify SSL", true);
```

Comment:
SSL property must have capital `L` at the end.

Reply:
Or a camel-case `Ssl`

Reply:
Good 🔍 I fixed the typo